arch/arm/soc/st_stm32/stm32l4/dts.fixup

```diff
@@ -74,6 +74,14 @@
 #define FLASH_DEV_BASE_ADDRESS		        ST_STM32L4_FLASH_CONTROLLER_40022000_BASE_ADDRESS
 #define FLASH_DEV_NAME			        ST_STM32L4_FLASH_CONTROLLER_40022000_LABEL
 
+#if defined(CONFIG_SOC_STM32L432XC)
```

Comment:
While, I don't see another way to propose the 2 variants of USB in fixup file (other than put them back in boards, which is not wanted), I'm not quite satisfied with this solution.
Main point is that if one wants to add a new SOC, he'll have to mess with this, while .fixup files should be as transparent as possible.

Can you put a comment to explain we're adding USB configuration depending on supported USB (USB/OTGFS), and one should add new SoC depending on supported USB?

I hope we should get rid of these .fixup files in near future, but meanwhile we should document any "tricks".

Reply:
I get your point that it will make support for more SoCs less easy. What about this then:

```
#if defined(ST_STM32_USB_40006800_BASE_ADDRESS)
...
#endif

#if defined(ST_STM32_OTGFS_50000000_BASE_ADDRESS)
...
#endif
```

I have tested it works for a nucleo_l432kc, and compiled tested for the stm32l476g_disco. As long as there is a single USB device on the SoC (which seems to be fine for STM32L4 family), that should work.

(BTW, what about enabling support for USB in stm32l476g_disco.dts and stm32l496g_disco.dts. That would make this kind of changes easier to test)

Reply:
Sounds like a nice alternative to me. Thanks

Reply:
> (BTW, what about enabling support for USB in stm32l476g_disco.dts and stm32l496g_disco.dts. That would make this kind of changes easier to test)

Yeah sure. This should be done as part of #7151, but it is welcome to start now